identity-account/src/account/account.rs

```diff
@@ -247,54 +276,66 @@ impl Account {
     &self,
     old_state: &IdentityState,
     new_state: &IdentityState,
+    fragment: &Option<String>,
     document: &mut IotaDocument,
   ) -> Result<()> {
-    if self.chain_state().is_new_identity() {
-      let method: &IotaVerificationMethod = new_state.document().default_signing_method()?;
-      let location: KeyLocation = new_state.method_location(
-        method.key_type(),
-        // TODO: Should be a fatal error.
-        method.id().fragment().ok_or(Error::MethodMissingFragment)?.to_owned(),
-      )?;
-
-      // Sign the DID Document with the current capability invocation method
-      new_state
-        .sign_data(self.did(), self.storage(), &location, document)
-        .await?;
+    let signing_document: &IotaDocument = if self.chain_state().is_new_identity() {
+      new_state.document()
     } else {
-      let method: &IotaVerificationMethod = old_state.document().default_signing_method()?;
-      let location: KeyLocation = new_state.method_location(
-        method.key_type(),
-        // TODO: Should be a fatal error.
-        method.id().fragment().ok_or(Error::MethodMissingFragment)?.to_owned(),
-      )?;
-
-      // Sign the DID Document with the previous capability invocation method
-      old_state
-        .sign_data(self.did(), self.storage(), &location, document)
-        .await?;
-    }
+      old_state.document()
+    };
+
+    let signing_method: &IotaVerificationMethod = match fragment {
+      Some(fragment) => signing_document
+        .resolve_method_with_scope(fragment, MethodScope::capability_invocation())
+        .ok_or(Error::DIDError(identity_did::Error::MethodNotFound))?,
+      None => signing_document.default_signing_method()?,
+    };
```

Comment:
This should also probably validate that the method is an `Ed25519VerificationKey2018`, similar to `IotaDocument::check_signing_key`. Right now a `MerkleKeyCollection2021` is rejected by `sign_data` simply because it is unimplemented in the `Account`.

This restriction is from the IOTA DID method specification: https://wiki.iota.org/identity.rs/specs/did/iota_did_method_spec#standardized-verification-method-types